src/ceph.rs

```diff
@@ -277,1337 +264,1422 @@ pub struct RadosVersion {
+        unsafe {
+            let ret_code = rados_stat(self.ioctx, object_name_str.as_ptr(), &mut psize, &mut time);
+            if ret_code < 0 {
+                return Err(RadosError::new(try!(get_error(ret_code as i32))));
+            }
+        }
+        Ok((psize, (UNIX_EPOCH + Duration::from_secs(time as u64))))
     }
-    let object_name_str = try!(CString::new(object_name));
-    let mut buffer: Vec<u8> = Vec::with_capacity(500);
 
-    unsafe {
-        let ret_code =
-            rados_tmap_get(ctx, object_name_str.as_ptr(), buffer.as_mut_ptr() as *mut c_char, buffer.capacity());
-        if ret_code == -ERANGE {
-            buffer.reserve(1000);
-            buffer.set_len(1000);
-            let ret_code =
-                rados_tmap_get(ctx, object_name_str.as_ptr(), buffer.as_mut_ptr() as *mut c_char, buffer.capacity());
+    /// Update tmap (trivial map)
+    pub fn rados_object_tmap_update(&self, object_name: &str, update: TmapOperation) -> RadosResult<()> {
+        self.ioctx_guard()?;
+        let object_name_str = try!(CString::new(object_name));
+        let buffer = try!(update.serialize());
+        unsafe {
+            let ret_code = rados_tmap_update(
+                self.ioctx,
+                object_name_str.as_ptr(),
+                buffer.as_ptr() as *const c_char,
+                buffer.len(),
+            );
             if ret_code < 0 {
                 return Err(RadosError::new(try!(get_error(ret_code as i32))));
             }
-        } else if ret_code < 0 {
-            return Err(RadosError::new(try!(get_error(ret_code as i32))));
         }
+        Ok(())
     }
-    match TmapOperation::deserialize(&buffer) {
-        IResult::Done(_, tmap) => Ok(tmap),
-        IResult::Incomplete(needed) => {
-            Err(RadosError::new(format!("deserialize of ceph tmap failed.
-            Input from Ceph was too small.  Needed: {:?} more bytes", needed)))
-        },
-        IResult::Error(e) => Err(RadosError::new(e.to_string())),
-    }
-}
 
-/// Execute an OSD class method on an object
-/// The OSD has a plugin mechanism for performing complicated operations on an
-/// object atomically.
-/// These plugins are called classes. This function allows librados users to
-/// call the custom
-/// methods. The input and output formats are defined by the class. Classes in
-/// ceph.git can
-/// be found in src/cls subdirectories
-pub fn rados_object_exec(
-    ctx: rados_ioctx_t, object_name: &str, class_name: &str, method_name: &str, input_buffer: &[u8],
-    output_buffer: &mut [u8]
-) -> RadosResult<()> {
-    if ctx.is_null() {
-        return Err(RadosError::new("Rados ioctx not created.  Please initialize first".to_string()));
-    }
-    let object_name_str = try!(CString::new(object_name));
-    let class_name_str = try!(CString::new(class_name));
-    let method_name_str = try!(CString::new(method_name));
+    // pub fn rados_object_tmap_put(ctx: rados_ioctx_t, object_name: &str) ->
```

Comment:
unnecessary comments?

Reply:
I think that was a function I was planning on implementing but never did.  I'm deleting it now.